test/e2e/compact_test.go

```diff
@@ -724,7 +724,7 @@ func testCompactWithStoreGateway(t *testing.T, penaltyDedup bool) {
 		operationMatcher, err := matchers.NewMatcher(matchers.MatchEqual, "operation", "get")
 		testutil.Ok(t, err)
 		testutil.Ok(t, c.WaitSumMetricsWithOptions(
-			e2emon.Equals(573),
+			e2emon.GreaterOrEqual(573),
```

Comment:
I think the idea of these assertions is to make sure the compactor is efficient and does not make more loops/API requests than necessary.

Reply:
I get the idea. But this is flaky because the number of operations depend heavily on time (could be a bug, I don't know), which becomes a hassle when processes are fighting for CPU time in CI or even if you run all e2e tests locally (because they will run in parallel). 

If I put a sleep on line 725, depending on how much time the sleep takes the assertion will fail because `thanos_objstore_bucket_operations_total` will have a completely different value. This test has to be fixed and made independent of time, otherwise it will stay flaky. But I do not have a clear picture of how to achieve it.

Reply:
Got it. In that case I am not sure what's the best option here, maybe we can use `LessThan(1000)` or some larger value to make sure the compactor does not go into infinite loops. If we have to use `GreaterThan`, we might as well remove the assertion.

Reply:
@fpetkovski I implemented a custom matcher to use in the assertion, to be able to assert the value is between 0 and 1000, instead of just checking one of the two bounds. This is not present yet in `efficientgo/e2emon`, so it lives in the `e2ethanos` package for now.

I've ran the test a few times locally with this configuration and it was so good that I am removing the skip on the penalty compactor test.

PTAL 🙇